mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -437,9 +437,14 @@ def set_roi_visibility_flags(self, roi_name: str, visible: bool) -> None:
         self.show_visible_spectrums()
 
     def show_visible_spectrums(self) -> None:
+        tof_data = self.presenter.model.tof_data
+        if tof_data is None or tof_data.size == 0:
+            spectrum_values = [v for v in self.spectrum_widget.spectrum_data_dict.values() if v is not None]
+            tof_data = np.arange(len(spectrum_values[0]) if spectrum_values else 0)
```

Comment:
I think we are okay to use `.size` here as `spectrum_values` should always be an np.ndarray. 

```Python
tof_data = np.arange(spectrum_values[0].size if spectrum_values else 0)
```